universal_binary.go

```diff
@@ -138,7 +137,7 @@ func (u *UniversalFile) Write(writer io.Writer) error {
 // IsUniversalMachoBinary returns true if this is a multi-architecture (universal) binary.
 func IsUniversalMachoBinary(reader io.ReaderAt) bool {
 	_, err := macho.NewFatFile(reader)
-	return !errors.Is(err, macho.ErrNotFat)
+	return err == nil
```

Comment:
Here is the main change.  Before binaries of other platforms would fail with: `&FormatError{0, "invalid magic number", nil}`